### PR TITLE
[MCC-1693] Removes is_valid check when reinserting 

### PR DESCRIPTION
**Pull request by mfaulk**
Soundtrack of this PR: [Be Together](https://www.youtube.com/watch?v=GZSsOEqgm0c)

### Motivation

TxManager should maintain a cache of well-formed transactions. Previously, it also checked for validity (which is funky and unnecessary) because other things incorrectly required the transactions in its cache to be valid.

### In this PR
* Removes is_valid check when reinserting a transaction into the cache.

### CODE REVIEW

**Review by rjwalters — APPROVED** [review]

**Review by eranrund — APPROVED** [review]
LGTM assuming this has passed deployment (unless you're confident we can skip that)

**Comment by mfaulk** [comment]
Thanks Eran, will deploy before merging

**Pull request merged**
